gateway/Dockerfile

```
@@ -48,10 +48,43 @@ COPY . .
 # Running env : build #
 #######################
 FROM alpine:3.20 AS running_env
-RUN apk add --no-cache curl ca-certificates bind-tools
+RUN apk add --no-cache curl ca-certificates bind-tools jq
 COPY --from=run_builder /usr/local/cargo/bin/osrd_gateway /usr/local/bin/osrd_gateway
 
 ARG OSRD_GIT_DESCRIBE
 ENV OSRD_GIT_DESCRIBE=${OSRD_GIT_DESCRIBE}
 
 CMD ["/usr/local/bin/osrd_gateway"]
+
+#####################
+# Build env : front #
+#####################
+FROM node:20-bookworm AS front_build
+WORKDIR /app
+
+# Build dependencies
+COPY --from=front_src package.json package-lock.json /app/
+RUN npm install --frozen-lockfile
+
+# Fill build time values
+ARG OSRD_GIT_DESCRIBE
+ENV VITE_OSRD_GIT_DESCRIBE=${OSRD_GIT_DESCRIBE}
+
+# Generate the licenses file and build
+COPY --from=front_src . /app
+RUN npm run generate-licenses && npm run build
```

Building the gateway crashed on my laptop with the error `FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory`.

I had to add this line here to solve the issue

```suggestion
ENV NODE_OPTIONS="--max_old_space_size=4096"
RUN npm run generate-licenses && npm run build
```